### PR TITLE
"migration" to c++14 : episode 2 - clang

### PR DESCRIPTION
**Pull request by pbougue**
after https://github.com/CanalTP/navitia/pull/2639

updated lz4 to v1.8.3 (latest release) as compiler was not correctly handled in version used
also removed a little clang warning on ref being a hidden copy

### CODE REVIEW

File: source/routing/next_stop_time.cpp [hunk 1]
```diff
@@ -437,7 +437,7 @@ CachedNextStopTime CachedNextStopTimeManager::CacheCreator::operator()(const Cac
 
 CachedNextStopTime::DtStFromJpp::DtStFromJpp(const vDtStByJpp& map) {
     size_t s = 0;
-    for (const auto& elt: map) {
+    for (const auto elt: map) {
```
Comment (TeXitoi) [inline source/routing/next_stop_time.cpp:440]: Why? It yields something as `std::pair<&T, &U>` ?

**Comment by kinnou02** [comment]
What was the problems? It didn't build with clang or there was some problems with gcc too?
Does this version of lz4 is rétrocompatible with the previous one? Will it load old .nav or do we need to rebuild them?

**Review by nonifier — APPROVED** [review]
:tada:

File: source/routing/next_stop_time.cpp (hunk 1 above)
Comment (pbougue) [reply source/routing/next_stop_time.cpp:440]: It uses those : https://github.com/CanalTP/utils/blob/master/idx_map.h#L130-L131

**Review by pbougue — COMMENTED** [review]
@kinnou02 : the problem for lz4 was the macro handling `deprecated` specifically depending on compiler, it was not handling clang c++14.
And I tested data.nav.lz4 generated without this PR : it works with a kraken from this PR.

**Review by TeXitoi — APPROVED** [review]

**Pull request merged**